src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -301,11 +298,14 @@ protected void verify(Checker checker,
 
             for (int i = 0; i < expected.length; i++) {
                 final String expectedResult = messageFileName + ":" + expected[i];
-                assertEquals("error message " + i, expectedResult, actuals.get(i));
+                assertWithMessage("error message %s", i)
+                        .that(actuals.get(i))
+                        .isEqualTo(expectedResult);
             }
 
-            assertEquals("unexpected output: " + lnr.readLine(),
-                    expected.length, errs);
+            assertWithMessage("unexpected output: " + lnr.readLine())
+                    .that(errs)
+                    .isEqualTo(expected.length);
```

Comment:
@shashwatj07 one more thought:

Couldn't the entire contents of this method be replaced by:

```java

        Map<String, List<String>> expectedViolations = new HashMap<>();
        expectedViolations.put(messageFileName, Arrays.asList(expected));
        verify(checker, processedFiles, expectedViolations);

```
?

Reply:
You're right. I've faced the same nuisance often times too. I would have made the changes myself if it weren't for this line here. What do you think?
https://github.com/checkstyle/checkstyle/blob/db5311e79c2f08fe4e2c46565370bf2c792ff684/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L273-L274

Reply:
It appears that this was added in https://github.com/checkstyle/checkstyle/pull/5236, but I do not see any discussion surrounding it. Considering that in both cases (for either `verify()` method that we are using), we are providing MORE information than was provided before, in a better format, I can't imagine why this would not be a good way to reuse existing code and improve debugging experience for maintainers.

FWIW, I think that it would be a good idea in this PR, and any future PR's for "Migrate to Truth..." where assertions are not simple, to temporarily modify a unit test so that it fails, then show a "before" and "after" in the PR description so that reviewers can compare between the output that Truth provides, and the old Hamcrest, JUnit, etc. assertions.

Reply:
done, sorry for the delay

Reply:
Resolved.